### PR TITLE
feat: Add separate eslint config for Vue2 code

### PR DESCRIPTION
**Pull request by crgwbr**

### CODE REVIEW

**Comment by jthomerson** [comment]
Nit: spelling of "separate" in commit / PR title

**Comment by crgwbr** [comment]
> Nit: spelling of "separate" in commit / PR title

Fixed. Someday I'll learn how to spell…

File: partials/vue.js [hunk 1]
```diff
@@ -0,0 +1,214 @@
+'use strict';
+const vueRulesBase = require('eslint-plugin-vue/lib/configs/base').rules,
+      vueRulesEssentials = require('eslint-plugin-vue/lib/configs/essential').rules,
+      vue3RulesEssentials = require('eslint-plugin-vue/lib/configs/vue3-essential').rules,
+      vueRulesStronglyRecommended = require('eslint-plugin-vue/lib/configs/strongly-recommended').rules,
+      vue3RulesStronglyRecommended = require('eslint-plugin-vue/lib/configs/vue3-strongly-recommended').rules,
+      vueRulesRecommended = require('eslint-plugin-vue/lib/configs/recommended').rules,
+      vue3RulesRecommended = require('eslint-plugin-vue/lib/configs/vue3-recommended').rules;
+
+const silvermineRules = {
```
Comment (yokuze) [inline partials/vue.js:10]: It looks like this object still contains Vue-3-specific rules, which means they are getting mixed into the Vue 2 config. For example, `vue/no-deprecated-v-is` applies to Vue 3 only.

We probably need separate object structures for base rules, Vue 2 rules, and Vue 3 rules.

**Review by yokuze — CHANGES_REQUESTED** [review]

**Comment by crgwbr** [comment]
Ready for your re-review @yokuze

**Review by yokuze — APPROVED** [review]
This LGTM. Thanks @crgwbr !

**Comment by yokuze** [comment]
@jthomerson please merge at will. @crgwbr can correct me if I'm wrong, but I don't think we will need a new NPM version until it's all tested and we are ready to tag a final version.

**Pull request merged**